include/libtorrent/torrent.hpp

```diff
@@ -164,10 +165,180 @@ namespace libtorrent
 		std::vector<char> restart_piece;
 	};
 
+	namespace flags {
```

Comment:
and enum class would not allow using the enums as indices in the bitfield without an explicit cast to ``int``, and since the flags shared names with a lot of other member functions in torrent, so they can't be in the same scope.